### PR TITLE
API documentation

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
This PR adds documentation to eox-tagging using rst format and sphinx

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+
+# -- Project information -----------------------------------------------------
+
+project = u'eox-tagging'
+copyright = u'2020, eduNEXT'
+author = u'eduNEXT'
+
+# The short X.Y version
+version = u'0.3'
```
Comment (morenol) [inline docs/source/conf.py:27]: Do we have to update this in each release?
 Could you change this to take the version from eox_tagging/__init__.py ? 
Or could you update bumpversion configuration to update this?

File: requirements/test.txt [hunk 2]
```diff
@@ -9,37 +9,51 @@ astroid==1.6.6            # via pylint
...
+coreapi==2.3.3            # via django-rest-swagger, openapi-codec
+coreschema==0.0.4         # via coreapi
+coverage==5.2             # via -r requirements/test.in
 django-crum==0.7.6        # via -r requirements/test.in
 django-fake-model==0.1.4  # via -r requirements/test.in
 django-oauth2-provider==0.2.6.1  # via -r requirements/test.in
+django-rest-swagger==2.2.0  # via -r requirements/test.in
 djangorestframework-oauth==1.1.0  # via -r requirements/test.in
-djangorestframework==3.9.4  # via -r requirements/base.txt, eox-core
+djangorestframework==3.9.4  # via -r requirements/base.txt, django-rest-swagger, eox-core
```
Comment (morenol) [inline requirements/test.txt:24]: I think that we need to add the constraint to use the same version used in ironwood.

File: Makefile [hunk 3]
```diff
@@ -53,3 +53,8 @@ run-quality-test: clean ## Run quality test.
 	$(TOX) pycodestyle ./eox_tagging
 	$(TOX) pylint ./eox_tagging --rcfile=./setup.cfg
 	$(TOX) isort --check-only --recursive --diff ./eox_tagging
+
+build-docs:
+	pip install sphinx
```
Comment (morenol) [inline Makefile:58]: @mariajgrimaldi I think that we should add a requirements/docs.in and a requirements/docs.txt files with the packages needed to generate the documentation, and use that here. We also would need to update the `upgrade` target to generate the  requirements/docs.txt file

File: docs/source/conf.py (hunk 1 above)
Comment (mariajgrimaldi) [reply docs/source/conf.py:27]: Yes! I'll change it

File: requirements/test.txt (hunk 2 above)
Comment (mariajgrimaldi) [reply requirements/test.txt:24]: Ok I'll add the constraint

File: Makefile (hunk 3 above)
Comment (mariajgrimaldi) [reply Makefile:58]: That would be better! 
The upgrade part is something like: 
$(PIP_COMPILE) -o requirements/docs.txt requirements/docs.in right?

Comment (morenol) [reply Makefile:58]: yes

File: requirements/test.txt (hunk 2 above)
Comment (morenol) [reply requirements/test.txt:24]: Run `make upgrade` again in order to update this

**Comment by mariajgrimaldi** [comment]
> do sphinx use the docstrings to generate the documentation ?
No, it uses the rst files defined in docs/

**Comment by mariajgrimaldi** [comment]
> do sphinx use the docstrings to generate the documentation ?

No, it uses the rst files defined in docs/

**Pull request merged**